app/models/grda_warehouse/hud/project.rb

```diff
@@ -329,10 +330,6 @@ def serves_children?
       # END_ACL
     end
 
-    def can?(user, permission: :can_view_projects)
-      self.class.viewable_by(user, permission: permission).where(id: id).exists?
```

Comment:
@eanders the underlying problem Martha identified was that  `viewable_by_entity(user, permission: x)` completely ignores the permission arg for non-ACL users.
https://github.com/greenriver/hmis-warehouse/blob/fe54601cc6d5a7f5a36a294e1a2347450f788c63/app/models/grda_warehouse/hud/project.rb#L304-L330

Is that a feature or a bug? I think it is misleading naming. I would expect it to check the permission in both cases.

Reply:
@eanders @gigxz basically this PR prioritizes non-ACL user ux over ACL user ux. Is that what we want to do?

Behavior on release-135:
- non-ACL users always see the "Client Locations" button if they can view the project, even without `can_view_project_locations` (incorrect)
- ACL users see the "Client Locations" button only if they have `can_view_project_locations` on this project (correct)

Behavior in this PR:
- non-ACL users only see the "Client Locations" button if they have some role granting them `can_view_project_locations` (correct)
- ACL users see the "Client Locations" button if they have `can_view_project_locations` anywhere, even if not in this project (incorrect, but we propose to plan to update this when we change the rest of this view to be ACL-compatible)

@eanders let me know if you think this makes sense to merge, or needs more work/thought?

Reply:
I agree with your assessment. I propose prioritizing the non-ACL user for `release-135`, since the majority of users are still role-based. And, addressing the underlying issue with viewable_by_entity in a future PR. That future PR can also address the other ACL issues in this view (e.g. always showing "edit" and "delete" without acl logic). IMO that should be a separate issue because it's for this entire view, not just the client location map link.

Reply:
Sorry, I probably won't have time to weigh in much for another couple hours.

Reply:
```
def can?(user, permission: :can_view_projects)
      return false if ! user.using_acls? && ! user.send("#{permission}?")

      self.class.viewable_by(user, permission: permission).where(id: id).exists?
    end
```

Reply:
Relevant HMIS pattern:
https://github.com/greenriver/hmis-warehouse/blob/bf9f28be3bfa1edc6e61188f1c08d39654066d56/drivers/hmis/app/models/hmis/user.rb#L74-L84

The `EntityAccessLoaderFactory` is a fancy thing Theron made that will resolve the relevant loader and entity fo the thing you're checking against. So if you want to check if someone has X permission for an Enrollment or a Service, it will return the relevant Project and ProjectAccessLoader. There's a lot of complex code there, and this was written in part to serve the performance needs of the GraphQL API, so I don't know if we need this level of complexity in the warehouse. 


In HMIS world, the view logic would be:
```
- if current_user.can_view_project_locations_for(@project)
  
```

Reply:
https://github.com/open-path/Green-River/issues/6785